src/network/websocket/WebSocketServer.cpp

```diff
@@ -91,6 +93,7 @@ bool SingleClientWSServer::addProtocol(std::unique_ptr<WebSocketProtocol> protoc
 
 void SingleClientWSServer::sendRawString(const std::string& protocolPath,
 										 const std::string& str) {
+	std::lock_guard lock(protocolMapMutex);
```

Comment:
I think this is also a case where we shouldn't hold the lock of the protocol map for the entire method. What if we changed `getProtocol()` to return an `std::optional<ProtocolData&>`, and everywhere else (where we guarantee that the entry exists) we immediately call `value()`, while here we can check if the optional is filled or not?